packages/plugin-errors/src/index.ts

```diff
@@ -79,38 +86,216 @@ export class PothosErrorsPlugin<Types extends SchemaTypes> extends BasePlugin<Ty
+      return resolver;
+    }
+
+    return async (source, args, context, info) => {
+      if (fieldConfig.kind === 'Subscription' && errorTypeMap.has(source as {})) {
+        return source;
+      }
+
+      try {
+        const result = (await resolver(source, args, context, info)) as never;
+
+        if (pothosItemErrors && result && typeof result === 'object' && Symbol.iterator in result) {
+          return yieldErrors(result, pothosItemErrors);
+        }
+
+        if (
+          pothosItemErrors &&
+          result &&
+          typeof result === 'object' &&
+          Symbol.asyncIterator in result
+        ) {
+          console.log(result, yieldAsyncErrors);
+          return yieldAsyncErrors(result, pothosItemErrors);
+        }
+
+        return result;
+      } catch (error: unknown) {
+        return wrapOrThrow(error, pothosErrors ?? []);
+      }
+    };
+  }
+
+  override wrapSubscribe(
+    subscribe: GraphQLFieldResolver<unknown, Types['Context'], object>,
+    fieldConfig: PothosOutputFieldConfig<Types>,
+  ): GraphQLFieldResolver<unknown, Types['Context'], object> | undefined {
+    const pothosErrors = fieldConfig.extensions?.pothosErrors as (typeof Error)[] | undefined;
+
+    if (!pothosErrors) {
+      return subscribe;
+    }
+
+    return (...args) => {
+      async function* yieldSubscribeErrors() {
+        try {
+          const iter = (await subscribe(...args)) as AsyncIterableIterator<unknown>;
+
+          if (!iter) {
+            return iter;
+          }
+
+          for await (const value of iter) {
+            yield value;
+          }
+        } catch (error: unknown) {
+          yield wrapOrThrow(error, pothosErrors ?? []);
+        }
+      }
+
+      return yieldSubscribeErrors();
+    };
+  }
+
+  createResultType(
+    parentTypeName: string,
+    fieldName: string,
+    type: TypeParam<Types>,
+    fieldType: PothosOutputFieldType<Types>,
+    errorOptions: ErrorFieldOptions<Types, TypeParam<Types>, unknown, false>,
+    directResultError: string,
+    defaultResultName: GetTypeName,
+    defaultUnionName: GetTypeName,
+    builderResultOptions: Normalize<
+      Omit<PothosSchemaTypes.ObjectTypeOptions<Types, {}>, 'interfaces' | 'isTypeOf'> & {
+        name?: GetTypeName;
+      }
+    > = {} as never,
+    builderUnionOptions: Normalize<
+      Omit<PothosSchemaTypes.UnionTypeOptions<Types>, 'resolveType' | 'types'> & {
+        name?: GetTypeName;
+      }
+    > = {} as never,
+  ) {
+    const errorBuilderOptions = this.builder.options.errors;
+    const { name: getResultName = defaultResultName, ...defaultResultOptions } =
+      builderResultOptions ?? {};
+    const { name: getUnionName = defaultUnionName, ...defaultUnionOptions } =
+      builderUnionOptions ?? {};
+
     const {
       types = [],
       result: {
         name: resultName = getResultName({
           parentTypeName,
-          fieldName: capitalize(fieldConfig.name),
+          fieldName,
```

Comment:
this was a bug, but probably needs a changeset since it has the potential to change type names in the schema if people were depending on this capitalization i custom name functions